backend/core/src/main/java/org/sonarsource/sonarlint/core/BindingClueProvider.java

```diff
@@ -141,53 +169,81 @@ private Set<String> matchConnections(BindingClue bindingClue, Set<String> eligib
   }
 
   @CheckForNull
-  private static ScannerProperties extractScannerProperties(ClientFile matchedFile) {
-    LOG.debug("Extracting scanner properties from {}", matchedFile);
-    var properties = new Properties();
+  private static BindingProperties extractSonarLintConfiguration(ClientFile sonarLintConfigurationFile) {
     try {
-      properties.load(new StringReader(matchedFile.getContent()));
-    } catch (Exception e) {
-      LOG.error("Unable to parse content of file '{}'", matchedFile, e);
+      var configuration = new Gson().fromJson(sonarLintConfigurationFile.getContent(), JsonObject.class);
+      var projectKey = configuration.get("projectKey");
+      var organization = configuration.get("sonarCloudOrganization");
+      var serverUrl = configuration.get("sonarQubeUri");
+      // Checking for PascalCase due to VS backward compatibility
+      if (projectKey == null || ((organization == null) == (serverUrl == null))) {
+        projectKey = configuration.get("ProjectKey");
+        organization = configuration.get("SonarCloudOrganization");
+        serverUrl = configuration.get("SonarQubeUri");
+      }
+      return new BindingProperties(projectKey != null ? projectKey.getAsString() : null,
+        organization != null ? organization.getAsString() : null,
+        serverUrl != null ? serverUrl.getAsString() : null,
+        true);
+    } catch (JsonSyntaxException e) {
```

Comment:
We are silently catching here, I think it won't be common to have a malformed JSON file, but we could at least print a log

Reply:
Agree